From: Ben Hutchings <ben@decadent.org.uk>
Subject: efivars: Remove check for 50% full on write
Date: Sat, 23 Mar 2013 02:18:42 +0000

On my EFI-booting system (AMI firmware/Asus board), the firmware does
not garbage-collect the variable store until it is rather more than
50% full, and it also updates a variable at every boot.  This check
means that variable writes are guaranteed to fail after the system has
booted more than a few hundred times.

Since pstore integration is now disabled by default in Debian, we will
not normally write that much data before rebooting and giving the
firmware a chance to garbage-collect the variable store.  Therefore,
until the check can be restricted to known-bad systems, it seems less
risky to disable it for now.

---
--- a/arch/x86/platform/efi/efi.c
+++ b/arch/x86/platform/efi/efi.c
@@ -114,7 +114,7 @@ static int __init setup_add_efi_memmap(c
 }
 early_param("add_efi_memmap", setup_add_efi_memmap);
 
-static bool efi_no_storage_paranoia;
+static bool efi_no_storage_paranoia = true;
 
 static int __init setup_storage_paranoia(char *arg)
 {
